microsoft/testsuites/dpdk/dpdkperf.py

```diff
@@ -0,0 +1,242 @@
+    category="performance",
+    description="""
+    This test suite is to validate DPDK performance
+    """,
+)
+class DpdkPerformance(TestSuite):
+    TIMEOUT = 12000
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: failsafe pmd
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_failsafe_dual_core(
+        self,
+        environment: Environment,
+        log: Logger,
+        variables: Dict[str, Any],
+    ) -> None:
+        self._run_dpdk_perf_test("failsafe", environment, log, variables)
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: failsafe pmd
+        Uses all available cores and default queues
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_failsafe_multi_core(
+        self,
+        environment: Environment,
+        log: Logger,
+        variables: Dict[str, Any],
+    ) -> None:
+
+        core_counts = [
+            n.tools[Lscpu].get_core_count() for n in environment.nodes.list()
+        ]
+
+        assert_that(core_counts).described_as(
+            "Nodes contain different core counts, DPDK Suite expects sender "
+            "and receiver to have same core count."
+        ).contains_only(core_counts[0])
+
+        self._run_dpdk_perf_test(
+            "failsafe", environment, log, variables, use_cores=core_counts[0]
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: failsafe pmd
+        uses multiple cores and tx/rx queues
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_failsafe_multi_queue(
+        self,
+        environment: Environment,
+        log: Logger,
+        variables: Dict[str, Any],
+    ) -> None:
+
+        self._run_dpdk_perf_test(
+            "failsafe", environment, log, variables, use_queues=True
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: netvsc pmd
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_netvsc_dual_core(
+        self,
+        environment: Environment,
+        log: Logger,
+        variables: Dict[str, Any],
+    ) -> None:
+        self._run_dpdk_perf_test("netvsc", environment, log, variables)
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: netvsc pmd
```

Comment:
The description is the same as above. They should describe differences among test cases.